### PR TITLE
[*/2018b] add patch to fix building Qt5.10.1 against newer glibc

### PR DESCRIPTION
**Pull request by branfosj**
(created using `eb --new-pr`)

* [x] #13785

### CODE REVIEW

File: easybuild/easyconfigs/q/Qt5/Qt5-5.10.1_glibc-fix.patch [hunk 1]
```diff
@@ -0,0 +1,26 @@
+Fix Qt5.10.1 to build against newer qlibc.
```
Comment (boegel) [inline easybuild/easyconfigs/q/Qt5/Qt5-5.10.1_glibc-fix.patch:1]: @branfosj Let's fix the `qlibc` -> `glibc` typo here? ;)

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/q/Qt5/Qt5-5.10.1_glibc-fix.patch (hunk 1 above)
Comment (branfosj) [reply easybuild/easyconfigs/q/Qt5/Qt5-5.10.1_glibc-fix.patch:1]: I made the same typo in the subject line :(

However, I hit a further issue in the build:
```
FAILED: gn
/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/Python/2.7.15-foss-2018b/bin/python2 "../../../../build/toolchain/gcc_link_wrapper.py" --output="./gn" -- /rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/GCCcore/7.3.0/bin/g++ -Wl,--fatal-warnings -fPIC -Wl,-z,noexecstack -Wl,-z,now -Wl,-z,relro -Wl,-z,defs -Wl,--no-as-needed -lpthread -Wl,--as-needed -fuse-ld=gold -B../../../../third_party/binutils/Linux_x64/Release/bin -m64 -Wl,-O1 -Wl,--gc-sections -Wl,-rpath-link=. -Wl,--disable-new-dtags -o "./gn" -Wl,--start-group @"./gn.rsp"  -Wl,--end-group   -ldl -lpthread -lrt
/lib/../lib64/crti.o(.gnu.build.attributes+0x14): warning: relocation refers to discarded section
/lib/../lib64/crti.o(.gnu.build.attributes+0x1c): warning: relocation refers to discarded section
/lib/../lib64/crtn.o(.gnu.build.attributes+0x14): warning: relocation refers to discarded section
/lib/../lib64/crtn.o(.gnu.build.attributes+0x1c): warning: relocation refers to discarded section
/lib/../lib64/crt1.o(.gnu.build.attributes.hot+0x18): warning: relocation refers to discarded section
/lib/../lib64/crt1.o(.gnu.build.attributes.hot+0x20): warning: relocation refers to discarded section
/lib/../lib64/crt1.o(.gnu.build.attributes.unlikely+0x18): warning: relocation refers to discarded section
/lib/../lib64/crt1.o(.gnu.build.attributes.unlikely+0x20): warning: relocation refers to discarded section
/lib/../lib64/crt1.o(.gnu.build.attributes.startup+0x18): warning: relocation refers to discarded section
/lib/../lib64/crt1.o(.gnu.build.attributes.startup+0x20): warning: relocation refers to discarded section
/lib/../lib64/crt1.o(.gnu.build.attributes.exit+0x18): warning: relocation refers to discarded section
/lib/../lib64/crt1.o(.gnu.build.attributes.exit+0x20): warning: relocation refers to discarded section
/lib/../lib64/crt1.o(.gnu.build.attributes.hot+0x18): warning: relocation refers to discarded section
/lib/../lib64/crt1.o(.gnu.build.attributes.hot+0x20): warning: relocation refers to discarded section
/lib/../lib64/crt1.o(.gnu.build.attributes.unlikely+0x18): warning: relocation refers to discarded section
/lib/../lib64/crt1.o(.gnu.build.attributes.unlikely+0x20): warning: relocation refers to discarded section
/lib/../lib64/crt1.o(.gnu.build.attributes.startup+0x18): warning: relocation refers to discarded section
/lib/../lib64/crt1.o(.gnu.build.attributes.startup+0x20): warning: relocation refers to discarded section
/lib/../lib64/crt1.o(.gnu.build.attributes.exit+0x18): warning: relocation refers to discarded section
/lib/../lib64/crt1.o(.gnu.build.attributes.exit+0x20): warning: relocation refers to discarded section
/usr/lib64/libc_nonshared.a(elf-init.oS)(.gnu.build.attributes.hot+0x18): warning: relocation refers to discarded section
/usr/lib64/libc_nonshared.a(elf-init.oS)(.gnu.build.attributes.hot+0x20): warning: relocation refers to discarded section
/usr/lib64/libc_nonshared.a(elf-init.oS)(.gnu.build.attributes.unlikely+0x18): warning: relocation refers to discarded section
/usr/lib64/libc_nonshared.a(elf-init.oS)(.gnu.build.attributes.unlikely+0x20): warning: relocation refers to discarded section
/usr/lib64/libc_nonshared.a(elf-init.oS)(.gnu.build.attributes.startup+0x18): warning: relocation refers to discarded section
/usr/lib64/libc_nonshared.a(elf-init.oS)(.gnu.build.attributes.startup+0x20): warning: relocation refers to discarded section
/usr/lib64/libc_nonshared.a(elf-init.oS)(.gnu.build.attributes.exit+0x18): warning: relocation refers to discarded section
/usr/lib64/libc_nonshared.a(elf-init.oS)(.gnu.build.attributes.exit+0x20): warning: relocation refers to discarded section
/rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/binutils/2.30-GCCcore-7.3.0/bin/ld.gold: error: treating warnings as errors
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
Command '['/dev/shm/build-branfosj-admin/branfosj-admin-up/Qt5/5.10.1/foss-2018b/qt-everywhere-src-5.10.1/qtwebengine/src/3rdparty/ninja/ninja', '-C', '/dev/shm/build-branfosj-admin/branfosj-admin-up/Qt5/5.10.1/foss-2018b/qt-everywhere-src-5.10.1/qtwebengine/src/3rdparty/chromium/tools/gn/out/Release', 'gn']' returned non-zero exit status 1
Building gn manually in a temporary directory for bootstrapping...
Building gn using itself to out/Release...
Project ERROR: GN build error!
```

Comment (branfosj) [reply easybuild/easyconfigs/q/Qt5/Qt5-5.10.1_glibc-fix.patch]: Spelling corrected in e7badbb

Build bug fixed by #13785

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0211u08b.bear.cluster - Linux RHEL 8.3, x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.6.8
See https://gist.github.com/70c2c611792249248c00a4445284c1c6 for a full test report.

**Comment by branfosj** [comment]
@boegelbot please test @ generoso
CORE_CNT=16

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 39 out of 39 (2 easyconfigs in total)
bber0501u03a.bb2.cluster - Linux RHEL 8.3, x86_64, Intel(R) Xeon(R) CPU E5-2690 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/4a06c3a605036b4d7486ee8423519d40 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 14 out of 15 (3 easyconfigs in total)
node3133.skitty.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/5d24e655ee7816016ab675328b50c6c5 for a full test report.

**Comment by branfosj** [comment]
So, this fixes the build issue for Qt5 5.10.1 foss and fosscuda versions. There are more issues with intel / 2018b. I do not have that available to test / debug.

**Comment by boegel** [comment]
See #8003 w.r.t. the failing build with `intel/2018b`.

We never backported the `binutils` patch to `binutils/2.30-GCCcore-7.3.0` used there, so no surprise the issue still stands.

I won't let that block this PR though, the effort required to fix that problem is probably no longer worth it...

**Review by boegel — APPROVED** [review]

**Pull request merged**